go/vt/vtgate/vstream_manager.go

```diff
@@ -401,6 +412,13 @@ func (vs *vstream) sendEvents(ctx context.Context) {
 				})
 				return
 			}
+		case <-vs.streamLivenessTimer.C:
+			msg := fmt.Sprintf("vstream failed liveness checks as there was no activity, including heartbeats, within the last %v", livenessTimeout)
+			log.Infof("Error in vstream: %s", msg)
+			vs.once.Do(func() {
+				vs.setError(vterrors.New(vtrpcpb.Code_UNAVAILABLE, msg), "vstream is fully throttled or otherwise hung")
```

Comment:
Will this include the keyspace/shard/tablet in the error message?

Reply:
This would only fire if it applies to all shards in the VStream.